tests/Run-UnitTests.ps1

```diff
@@ -93,207 +162,132 @@ function DeployTests($testList)
-                }
-            }
-        }
-        
-        if ($testsuite.disabled -ne 0) {
-            foreach ($testcase in $testsuite.testcase) {
-                if ($testcase.status -ne "run") {
-                    $disabledMessageArray += [tuple]::Create([tuple]::Create($xmlTuple.Item1, $testsuite.name, $testcase.Name), 0)
-                }
-            }
-        }
+# Decide where the WTL output files will live
+if ($WTLOutputDirectory -ne [string]$null)
+{
+    $WTTLogFullPath = Join-Path $WTTLogPath wttlog.dll 
+    Copy-Item $WTTLogFullPath -Destination $TestSrcDirectory -Force
 
+    if ($TargetingDevice)
+    {
+        $outputLocalName = Join-Path -Path $WTLOutputDirectory -ChildPath $outputFileName
+        $outputRemoteName = Join-Path -Path $TestDstDirectory -ChildPath $outputFileName
+        $argList += " /logFile:$outputRemoteName /enableWttLogging"
     }
-}
-
-
-Write-Host "-----------------------------------------------------"
-Write-Host "SUMMARY Passed: " -NoNewLine
-Write-Host ($testCount - $failureCount - $disabledCount) -foregroundcolor "Green" -NoNewLine
-Write-Host  " Failed: " -NoNewLine 
-Write-Host ($failureCount) -foregroundcolor "Red" -NoNewLine
-Write-Host  " Disabled: " -NoNewLine 
-Write-Host ($disabledCount) -foregroundcolor "Yellow" -NoNewLine
-Write-Host  " Total: " -NoNewLine 
-Write-Host  $testCount
-Write-Host  ""
-
-if ($disabledMessageArray.length -ne 0) {
-    Write-Host  "DISABLED TESTS:" -foregroundcolor "Yellow"
-
-    foreach ($disabledMessage in $disabledMessageArray) {
-        Write-Host  "  " $disabledMessage.Item1.Item2 -foregroundcolor "Yellow" -NoNewLine 
-        Write-Host  "." -foregroundcolor "Yellow" -NoNewLine
-        Write-Host  $disabledMessage.Item1.Item3 -foregroundcolor "Yellow"
+    else
+    {
+        $outputLocalName = Join-Path -Path $WTLOutputDirectory -ChildPath $outputFileName
+        $argList += " /logFile:$outputLocalName /enableWttLogging"
     }
-    
-    Write-Host  ""
 }
 
-if ($crashingTestArray.length -ne 0) {
-    Write-Host  "PROBLEMS RUNNING MODULES:" -foregroundcolor "Red"
-
-    foreach ($crash in $crashingTestArray) {
-        Write-Host  "  " $crash -foregroundcolor "Red"
-    }
 
-    Write-Host  ""
+if($TestFilter -ne [string]$null)
+{
+    $argList += " /name:$TestFilter"
 }
 
-if ($failureMessageArray.length -ne 0) {
-    Write-Host  "FAILING TESTS:" -foregroundcolor "Red"
-
-    foreach ($errorMessage in $failureMessageArray) {
-        Write-Host  "  " $errorMessage.Item1.Item2 -foregroundcolor "Red" -NoNewLine 
-        Write-Host  "." -foregroundcolor "Red" -NoNewLine
-        Write-Host  $errorMessage.Item1.Item3 -foregroundcolor "Red" -NoNewLine
-        $reproLine = "  ( " + $errorMessage.Item1.Item1 + " --gtest_filter=" + $errorMessage.Item1.Item2 + "." + $errorMessage.Item1.Item3 + " )"
-        Write-Host  $reproLine -foregroundcolor "Red"
-    }
-}
+ExecTest $argList
 
-exit $failureCount + $crashingTestArray.length
+exit
```

Comment:
[https://winobjc.visualstudio.com/WinObjC/_build/explorer?_a=summary&definitionType=2&buildId=7825](https://winobjc.visualstudio.com/WinObjC/_build/explorer?_a=summary&definitionType=2&buildId=7825)

fails.  It has some errors from CG.


---
In reply to: [100258200](https://github.com/Microsoft/WinObjC/pull/1849#discussion_r100258200) [](ancestors = 100258200)